### ISSUE TITLE
Add Flower Baseline: [FedRS]

### ISSUE THREAD

**Issue by flydump** [adap/flower#4546]
### Paper

Xin-Chun Li, De-Chuan Zhan. FedRS: Federated Learning with Restricted Softmax for Label Distribution Non-IID Data (KDD'21)

### Link

https://dl.acm.org/doi/10.1145/3447548.3467254

### Maybe give motivations about why the paper should be implemented as a baseline.

FedRS (Federated Learning with Restricted Softmax) is a method to correct the negative effect of local training with missing classes by restricting the update of weights via a corrected softmax term. The main parameter is alpha (between 0.0 and 1.0), which determines the strength of this correction. If alpha=1.0, this is same as vanilla FedAvg. 

FedRS (100+ citations) is referenced in other papers exploring Federated Learning with non-iid datasets, such as [FedLC](https://arxiv.org/abs/2209.00189) (Federated Learning with Label Distribution Skew via Logits Calibration) and [FedConcat](https://arxiv.org/abs/2312.06290) (Exploiting Label Skews in Federated Learning with Model Concatenation). 

The plan is to reproduce the results for FedAvg and FedRS (with different alpha values) in Table 5 of the paper:
<img width="509" alt="Screenshot 2024-11-19 at 06 14 23" src="https://github.com/user-attachments/assets/66f11575-5c62-4c9d-b501-c9f368391c33">

### Is there something else you want to add?

I've found no mention of FedRS in PRs or in existing baselines. I've started the repro process.

### For first time contributors

- [x]  Read the [`first contribution` doc](https://flower.ai/docs/first-time-contributors.html)
- [x]  Complete the Flower tutorial
- [x]  Read the Flower Baselines docs to get an overview:
    - [x]  [How to use Flower Baselines](https://flower.ai/docs/baselines/how-to-use-baselines.html)
    - [x]  [How to contribute a Flower Baseline](https://flower.ai/docs/baselines/how-to-contribute-baselines.html)

### Prepare - understand the scope

- [X] Read the paper linked above
- [X] Decide which experiments you'd like to reproduce. The more the better!
- [X] Follow the steps outlined in [Add a new Flower Baseline](https://flower.ai/docs/baselines/how-to-contribute-baselines.html#add-a-new-flower-baseline).
- [X] You can use as reference [other baselines](https://github.com/adap/flower/tree/main/baselines) that the community merged following those steps.

### Verify your implementation

- [X] Follow the steps indicated in the `EXTENDED_README.md` that was created in your baseline directory
- [X] Ensure your code reproduces the results for the experiments you chose
- [X] Ensure your `README.md` is ready to be run by someone that is no familiar with your code. Are all step-by-step instructions clear?
- [X] Ensure running the formatting and typing tests for your baseline runs without errors.
- [X] Clone your repo on a new directory, follow the guide on your own `README.md` and verify everything runs.

**Comment by jafermarq** [comment]
Hey @flydump , it's great to see you want to include `FedRS` into Flower Baselines collection 🙌 ! I note the PR is in draft mode, do ping us when you'd like us to start the reviewing process. If you got any questions, feel free to ask too!!

**Comment by WilliamLindskog** [comment]
Hi @flydump, 

Could you please take a look at the comments made by @yan-gao-GY in the linked PR? Happy to help out if there is anything you need to discuss. 

Best regards
William

**Comment by WilliamLindskog** [comment]
Closing this PR together with issue as it has gone stale. We will migrate this to [Flower Disucss Topic](https://discuss.flower.ai/t/federated-baseline-contributions/602) for now.
